File: notepad-frontend/src/api/model/authdto.ts


interface loginRequestdto {
    username: string,
    password: string,

}

interface signInRequestdto {
    username: string,
    password: string,
    email: string,
    firstName: string,
    lastName: string,
    phoneNumber: string,

}


interface tokenResponcedto {
    accessToken: string,
    tokenType: string,
    refreshToken: string,
    expiresIn: number,
    expiresAt: string,
}

interface refreshTokenRequestdto {
    refreshToken: string,
}



export type { loginRequestdto, tokenResponcedto, signInRequestdto, refreshTokenRequestdto }

File: notepad-backend/Dto/AuthDto.cs
using System.Text.Json.Serialization;

namespace notepad_backend.Dto;



public class LoginRequestDto
{
    public string Username { get; set; } = string.Empty;
    public string Password { get; set; } = string.Empty;
}

public class RegisterRequestDto
{
    public string Username { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Password { get; set; } = string.Empty;
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? PhoneNumber { get; set; }
}

public class RefreshTokenRequestDto
{
    public string RefreshToken { get; set; } = string.Empty;
    public int AuthType { get; set; }
}

public class TokenResponseDto
{
    [JsonPropertyName("access_token")]
    public string AccessToken { get; set; } = string.Empty;

    [JsonPropertyName("refresh_token")]
    public string? RefreshToken { get; set; }

    [JsonPropertyName("expires_in")]
    public int ExpiresIn { get; set; }

    [JsonPropertyName("scope")]
    public string? Scope { get; set; }

    [JsonPropertyName("token_type")]
    public string TokenType { get; set; } = string.Empty;

    [JsonPropertyName("id_token")]
    public string? IdToken { get; set; }

    [JsonPropertyName("expires_at")]
    public DateTime ExpiresAt { get; set; }

    [JsonPropertyName("auth_type")]

    public int AuthType { get; set; }

}